hashintel/hash
# H-977, H-978: Allow assigning view/edit access to users and webs of users

## 🌟 What is the purpose of this PR?




This PR adds the ability to assign edit/view access to specific users and orgs.

## 🔗 Related links





- H-977
- H-978

## 🔍 What does this change?




See commits.

## Pre-Merge Checklist 🚀

### 🚢 Has this modified a publishable library?




This PR:


### 📜 Does this require a change to the docs?




The changes in this PR:


### 🕸️ Does this require a change to the Turbo Graph?




The changes in this PR:


## ⚠️ Known issues



- Currently modifying the permissions of a page does not mean that the blocks and linked entities in the page inherit the updated permissions (related to H-829)

## 🐾 Next steps



## 🛡 What tests cover this?



None

## ❓ How to test this?



⚠️ This PR requires backend changes to be applied to main, so using the Vercel deployment risks breaking the production environment

1. Go to a page or entity and click "Share" on the top right
2. Use the UI to invite users or orgs to the entity, and modify the permissions
3. Observe that there is a new "Shared with others" state in the status button once the entity has been shared

## 📹 Demo



https://github.com/hashintel/hash/assets/42802102/ef0894fe-5ada-4448-b848-e212529df4dc
